src/core/raster/qgsrasterattributetable.h

```diff
@@ -262,24 +307,116 @@ class CORE_EXPORT QgsRasterAttributeTable
     /**
      * Returns the \a value for \a row and \a column.
      */
-    QVariant value( const int row, const int column );
+    QVariant value( const int row, const int column ) const;
+
+    /**
+     * Returns the minimum value of the MinMax (thematic) or Min (athematic) column, returns NaN on errors.
+     */
+    double minValue( ) const;
+
+    /**
+     * Returns the maximum value of the MinMax (thematic) or Max (athematic) column, returns NaN on errors.
+     */
+    double maxValue( ) const;
+
+    /**
+     * Returns a row of data for the given \a matchValue or and empty row
+     * if there is not match.
+     */
+    QVariantList row( const double matchValue ) const;
+
+    /**
+     * Returns the classes for a thematic Raster Attribute Table, classified
+     * by \a classificationColumn, the default value of -1 makes the method guess
+     * the classification column based on the field usage.
+     */
+    QList<QgsRasterAttributeTable::MinMaxClass> minMaxClasses( const int classificationColumn  = -1 ) const;
+
+    /**
+     * Returns the color ramp for an athematic Raster Attribute Table
```

Comment:
is "athematic" a typo or deliberate?

Reply:
deliberate. Again, from GDAL: GRTT_ATHEMATIC, I'm open to suggestions.